packages/contracts/contracts/libs/v0.8.x/minter-libs/RAMLib.sol

```diff
@@ -0,0 +1,2923 @@
+                        startSlotIndex: uint16(
+                            currentLatestMintedBidSlotIndex - 1
+                        )
+                    });
+                    require(
+                        currentLatestMintedBidSlotIndex < NUM_SLOTS,
+                        "slot with bid not found"
+                    );
+                    // current bid is now the head of the linked list
+                    currentLatestMintedBidId = RAMProjectConfig_
+                        .headBidIdBySlot[currentLatestMintedBidSlotIndex];
+                }
+            }
+
+            // @dev minter is in State C, so bid must not have been minted or
+            // refunded due to scrolling logic of admin mint and refund
+            // functions available for use while in State C. The bid may have
+            // been previously settled, however.
+
+            // Mint bid and settle if not already settled
+            // @dev scrolling logic in State C ensures bid is not yet minted
+            _mintAndSettle({
+                projectId: projectId,
+                coreContract: coreContract,
+                projectPrice: projectPrice,
+                slotIndex: currentLatestMintedBidSlotIndex,
+                bidId: currentLatestMintedBidId,
+                minterFilter: minterFilter,
+                minterRefundGasLimit: minterRefundGasLimit
+            });
+
+            // increment num new tokens minted
+            unchecked {
+                ++numNewTokensMinted;
+            }
+        }
+
+        // finally, update auction metadata storage state from memoized values
+        // @dev safe to cast numNewTokensMinted to uint24
+        RAMProjectConfig_.numBidsMintedTokens += uint24(numNewTokensMinted);
+        // @dev safe to cast to uint32 because directly derived from bid ID
+        RAMProjectConfig_.latestMintedBidId = uint32(currentLatestMintedBidId);
+    }
+
+    /**
+     * @notice Function to mint and settle bid if not already settled.
+     * @param projectId Project ID to mint token on.
+     * @param coreContract Core contract address for the given project.
+     * @param projectPrice Price of a token for the given project.
+     * @param slotIndex Slot index of bid.
+     * @param bidId ID of bid to settle.
+     * @param minterFilter Minter filter contract address
+     * @param minterRefundGasLimit Gas limit to use when settling bid, prior to using fallback force-send to refund
+     */
+    function _mintAndSettle(
+        uint256 projectId,
+        address coreContract,
+        uint256 projectPrice,
+        uint256 slotIndex,
+        uint256 bidId,
+        IMinterFilterV1 minterFilter,
+        uint256 minterRefundGasLimit
+    ) internal {
+        // load project config
+        RAMProjectConfig storage RAMProjectConfig_ = getRAMProjectConfig({
+            projectId: projectId,
+            coreContract: coreContract
+        });
+        Bid storage bid = RAMProjectConfig_.bids[bidId];
```

Comment:
should this list an assumption in the natspec of the function that a non-null bidder address has already been checked?

Reply:
Added! The check in `adminArtistAutoMintTokensToWinners` isn't as explicit as the one in `directMintTokensToWinners`, it's checking that there is a slot with bid returned by `_getMaxSlotWithBid` but not explicitly checking the bidder address is not null.. Do you think it's redundant to check if the bidder address is not null before calling `_mintAndSettle` in that function?

Reply:
yeah, I think the dev comment in `function adminArtistAutoMintTokensToWinners` that says `// @dev scrolling logic in State C ensures bid is not yet minted` is sufficient to not need a check, so this lgtm! 🙏

Reply:
actually, maybe that comment should say: `// @dev scrolling logic in State C ensures bid **exists** is not yet minted`?

Reply:
Good call - that's better, updated!